### DISCUSSION TITLE
请教一下，怎么配置才能支持wasm呢？

### DISCUSSION THREAD

**Discussion by kurisu994** [web-infra-dev/modern.js#4144] [Q&A]
项目是modernjs+ rspack，wasm是用wasm-bindgen编译打成npm包的，引入项目会直接提示 ：
Uncaught (in promise) TypeError:  xxx_bgwasm.__wbindgen_start is not a function 
文档上也没找到哪里可以配置 asyncWebAssembly 属性

**Comment by chenjiahan** [comment]
Wasm 是默认支持的，参考 [引用 Wasm 资源](https://modernjs.dev/builder/guide/basic/wasm-assets.html)。

如果有问题的话，可以执行 npx modern upgrade 升级试一下

**Reply by kurisu994** [reply]
我指的不是直接引用wasm资源，我已经使用[wasm-pack](https://github.com/rustwasm/wasm-pack)编译成npm包，我并不能正常的使用它。
如果我使用webpack来构建的话是可以正常运行的

**Reply by chenjiahan** [reply]
或许可以提供一个复现 repo，这样我们可以深入排查下。

另外上面提到的 `asyncWebAssembly` 是默认开启的，你也可以通过 tools.bunderChain 来修改 webpack / Rspack 配置，参考 https://modernjs.dev/builder/api/config-tools.html#toolsbundlerchain

**Reply by kurisu994** [reply]
好的，这是demo代码地址 https://gitee.com/makisek/modern-demo

**Reply by chenjiahan** [reply]
我验证了一下这应该属于 Rspack 的 bug，你可以到 Rspack 仓库下新建一个 Issue 哈：https://github.com/web-infra-dev/rspack
